core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -111,7 +111,7 @@ class ReplicaAlterLogDirsThread(name: String,
   override def processPartitionData(topicPartition: TopicPartition,
                                     fetchOffset: Long,
                                     partitionData: PartitionData[Records]): Option[LogAppendInfo] = {
-    val partition = replicaMgr.nonOfflinePartition(topicPartition).get
+    val partition = replicaMgr.onlinePartition(topicPartition).get
```

Comment:
It seems like we should be using `getPartitionOrException` here. Similarly in `ReplicaFetcherThread`.

Reply:
Yes, this appears to be an existing issue.  I fixed it in 4 places.